### ISSUE TITLE
Add location and altitude accuracy to OData feed

### ISSUE THREAD

**Issue by florianm** [opendatakit/central#73]
Bulk exporting to CSV exports location accuracy (which is very useful), but not altitude accuracy (which is mostly too imprecise). It would be great if the OData feed would export at least the location accuracy - currently it exports lat/lon and another (unnamed) integer.

**Comment by issa-tseng** [comment]
so, the location export format for CSV is [WKT](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry), which is a much better standard for inline text like CSV and is rapidly becoming the format of choice for these kinds of applications.

so, as you can see the third number is the altitude (`(x, y, z)`). (though now looking at it, it looks like i was supposed to add a `z` after the alt. oops.)

unfortunately, accuracy is not part of the WKT spec, and we debated how to include the accuracy information before deciding to deal with it later. generating a synthetic column for the information probably makes the most sense, but it also involves risks like fieldname collisions.

**Comment by issa-tseng** [comment]
as for OData, i think we can export the accuracy as a geojson property. i'll look into that.

**Comment by florianm** [comment]
Considering WKT makes a ton of sense. 

From a business side, the location (lat/lon) accuracy is highly sought after. My data analysts always ask for the source of the location (GPS, hand drawn marker on paper map - and at what scale, or hand-drawn on a leaflet map widget) and also the accuracy.

Could name collisions be avoided if you'd prefix the horizontal accuracy with the location field name (considering that's unique)? What would happen if length(location field name) + length("accuracy") > max field name length?

**Comment by issa-tseng** [comment]
okay, i need to wake up properly before answering these tickets. it sounds like the CSV output is fine, which i did not correctly understand earlier. it already exports the altitude _and_ the accuracy (but _not_ the altitude accuracy, which i got confused by).

the odata problem should be easy to solve, since geojson allows arbitrary embedded metadata. so in that case i will simply call it `accuracy`, unless you feel there is a strong case for calling it something like `XYAccuracy` instead.

**Comment by issa-tseng** [comment]
(as for altitude accuracy, i think part of my confusion is that to my knowledge (and looking through the Collect source seems to confirm this) we don't do altitude accuracy at all. perhaps this will have an effect on your answer on `accuracy` being the name.)

**Comment by florianm** [comment]
Yep that would work! +1 for including "accuracy" anywhere you see fit in the OData feed. Found this: https://github.com/geojson/geojson-spec/issues/4

Sorry, yes, the CSV is fine (with acc but without alt acc). I think until generally the hardware spits out more reliable altitude accuracy, we will not really need altitude accuracy at all. 
We could probably assume that the observer is close enough to the earth surface in most cases to derive a more accurate altitude from a DEM if needed.
